### PR TITLE
fix(api): apply tenant primary override to service instance defaults

### PR DESCRIPTION
**Pull request by ashleythedeveloper**
This PR fixes service instance listing so that system defaults show `isPrimary: false` when the requesting tenant has their own primary for the same serviceType. Previously, system defaults always displayed `isPrimary: true` regardless of whether a tenant had set their own primary, making it appear that two services of the same type were both primary. This mirrors the existing DID default override pattern already in use for the `/dids` endpoints.

The override is applied at read time in both `getServiceInstanceById` (single fetch) and `listServiceInstances` (list fetch). The tenant primaries query runs alongside the main data query in `Promise.all`, so it adds no sequential latency. The resolution chain (`getInstanceByResolution`) was already correct and is unchanged.

## Test plan
- [x] Unit tests: override applied in `getServiceInstanceById` when tenant has own primary
- [x] Unit tests: override skipped when tenant has no primary for same serviceType
- [x] Unit tests: `listServiceInstances` overrides per-serviceType (VC overridden, IDR kept)
- [x] Unit tests: pagination edge case, tenant primary on different page still triggers override
- [x] E2E tests: system default shows primary when no tenant primary exists
- [x] E2E tests: system default overridden after tenant creates their own primary
- [x] E2E tests: other serviceTypes unaffected
- [x] E2E tests: `GET /services/:id` applies override
- [x] E2E tests: PATCH `isPrimary: false` restores system default
- [x] E2E tests: PATCH `isPrimary: true` re-overrides system default
- [x] E2E tests: DELETE tenant primary restores system default
- [x] All 30 repository unit tests passing

### CODE REVIEW

File: packages/reference-implementation/src/lib/prisma/repositories/service-instance.repository.ts [hunk 1]
```diff
@@ -58,17 +58,42 @@ export async function createServiceInstance(input: CreateServiceInstanceInput):
...
+): Promise<ServiceInstance | null> {
+  if (!instance || instance.tenantId !== SYSTEM_TENANT_ID || !instance.isPrimary) return instance;
+
+  const tenantPrimary = await prisma.serviceInstance.findFirst({
+    where: {
+      tenantId,
+      serviceType: instance.serviceType,
+      isPrimary: true,
+    },
+  });
```
[the remark this answers is not in the document]
Comment (ashleythedeveloper) [reply packages/reference-implementation/src/lib/prisma/repositories/service-instance.repository.ts:79]: Done in d2a5d3c — added select: { id: true }.

File: packages/reference-implementation/src/lib/prisma/repositories/service-instance.repository.ts [hunk 2]
```diff
@@ -92,16 +117,33 @@ export async function listServiceInstances(
...
+      data: data.map((i) =>
+        i.tenantId === SYSTEM_TENANT_ID && i.isPrimary && tenantPrimaryTypes.has(i.serviceType)
+          ? { ...i, isPrimary: false }
+          : i,
+      ),
+      total,
+    };
+  }
+
   return { data, total };
```
[the remark this answers is not in the document]
Comment (ashleythedeveloper) [reply packages/reference-implementation/src/lib/prisma/repositories/service-instance.repository.ts]: Done in d2a5d3c — removed the guard, map is a no-op when the set is empty.
